model/category_gibbs_lda.py

```diff
@@ -0,0 +1,146 @@
+import pickle
+from sklearn.model_selection import train_test_split
+import time
+from typing import List
+
+import numpy as np
+from tqdm import tqdm
+
+from preprocess.preprocessing import prepro_file_load
+import coherence
+
+
+def random_initialize(documents):
+    """
+    Randomly initialisation of the word topics
+    :param documents: a list of documents with their word ids
+    :return: word_topic_assignment,
+    """
+    print("Random Initilization")
+    category_topic = np.zeros([num_categories, num_topics]) + alpha
+    topic_word = np.zeros([num_topics, M]) + beta
+    topic_c = np.zeros([num_topics]) + M * beta
+    wt_assignment = []
+    for d, doc in tqdm(list(enumerate(documents))):
+        curr_doc = []
+        cat = doc2category[d]
+        for w in doc:
+            pz = np.divide(np.multiply(category_topic[cat, :], topic_word[:, w]), topic_c)
+            z = np.random.multinomial(1, pz / pz.sum()).argmax()
+            curr_doc.append(z)
+            category_topic[cat, z] += 1
+            topic_word[z, w] += 1
+            topic_c[z] += 1
+        wt_assignment.append(curr_doc)
+    return wt_assignment, category_topic, topic_word, topic_c
+
+
+def gibbs_sampling(documents: List[np.ndarray],
+                   cat_topic: np.ndarray,
+                   topic_word: np.ndarray,
+                   topic_count: np.ndarray,
+                   word_topic_assignment: List[List[int]]):
+    """
+    Takes a set of documents and samples a new topic for each word within each document.
+    :param word_topic_assignment: A list of documents where each index is the given words topic
+    :param topic_count: the number of the times each topic is used
+    :param documents: a list of documents with their word ids
+    :param cat_topic: a matrix describing the number of times each topic within each document
+    :param topic_word: a matrix describing the number of times each word within each topic
+    """
+    for d_index, doc in enumerate(documents):
+        for w_index, word in enumerate(doc):
+            # Find the topic for the given word a decrease the topic count
+            topic = word_topic_assignment[d_index][w_index]
+            c_index = doc2category[d_index]
```

Comment:
```suggestion
        c_index = doc2category[d_index]
        for w_index, word in enumerate(doc):
            # Find the topic for the given word a decrease the topic count
            topic = word_topic_assignment[d_index][w_index]
```
Rykker `c_index` ud i doc løkken